### PR TITLE
Migrate `interdependent-c-libraries`, `compiler-rt-works-on-mingw` and `incr-foreign-head-span` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

try-job: aarch64-apple
try-job: armhf-gnu
try-job: test-various
try-job: x86_64-mingw
try-job: x86_64-msvc
try-job: x86_64-gnu-llvm-18

### CODE REVIEW

**Comment by Oneirical** [comment]
@rustbot review
Please run bors try, ignore restored in `interdependent-c-libraries`.

**Review by jieyouxu — APPROVED** [review]
Thanks, the changes look reasonable, just some minor nits.

File: tests/run-make/compiler-rt-works-on-mingw/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// that compilation and execution with it are successful.
+// See https://github.com/rust-lang/rust/pull/29478
+
+//@ only-windows-gnu
+
+use run_make_support::{cxx, is_msvc, llvm_ar, run, rustc, static_lib_name};
+
+fn main() {
+    cxx().input("foo.cpp").arg("-c").out_exe("foo.o").run();
+    llvm_ar().obj_to_ar().output_input(static_lib_name("foo"), "foo.o").run();
```
[the remark this answers is not in the document]
Comment (Oneirical) [reply tests/run-make/compiler-rt-works-on-mingw/rmake.rs:12]: Review comments mostly implemented, except this one. Do you mean:

`let foo_object = PathBuf::from("foo.o");` and then using that? Why is this better than the current form?

[the remark this answers is not in the document]
Comment (jieyouxu) [reply tests/run-make/compiler-rt-works-on-mingw/rmake.rs:12]: Actually please disregard, the current form is fine.

**Review by jieyouxu — APPROVED** [review]
Thanks!

**Comment by jieyouxu** [comment]
This needs a rebase for the tidy list, r=me after that.
@bors delegate+

**Pull request merged**